docs/tutorials/vagrant-discovery.rst

```diff
@@ -0,0 +1,376 @@
+
+.. image:: ../_static/clear_mongodb.jpg
+     :align: center 
+
+**2. For demo environment**
+
+- stop rackhd
+.. code::
+
+    vagrant@rackhd:~$ sudo service rackhd stop
+
+- Clean data base by typing the following command
+
+.. code::
+
+   vagrant@rackhd:~$ mongo pxe
+
+- When the ">" prompt appears, type the following command
+
+.. code::
+
+   > db.dropDatabase()
+
+- Press Ctrl+D to exit. Mongo will display ‘bye’, and then exit
+
+Restart RackHD
+~~~~~~~~~~~~~~
+
+**1. For development environment**
+
+- Start RackHD again by typing the following command.
+
+.. code::
+
+    vagrant@rackhd:~$ sudo su
+    root@rackhd:/home/vagrant# pm2 start /home/vagrant/rackhd-pm2-config.yml
+
+- To use RackHD conveniently, the auth function of RackHD can be closed. Just edit ``/opt/monorail/config.json``, set 16 line `"authEnabled": true` to `"authEnabled": false`. Then restart RackHD service.
+
+.. code::
+
+    root@rackhd:/home/vagrant# pm2 restart /home/vagrant/rackhd-pm2-config.yml
+
+-  Wait several seconds, and then type the following RackHD restful API to ensure RackHD is started and is running successfully.
+
+.. code::
+
+    root@rackhd:/home/vagrant# curl localhost:8080/api/2.0/nodes
+
+-  The following message appears when RackHD is booting and not ready. The process takes several seconds. You can retry after 10 seconds.
+
+.. code::
+
+  curl: (7) Failed to connect to localhost port 8080: Connection refused
+
+-  When the API returns "[]"(no nodes discovered so far) RackHD software stacks are running successfully.
+ 
+**2. For demo environment**
+
+- Start RackHD again by typing the following command.
+
+.. code::
+  
+    vagrant@rackhd:~$ sudo service rackhd start
+
+- To use RackHD conveniently, the auth function of RackHD can be closed. Just edit ``/opt/monorail/config.json``, set 16 line `"authEnabled": true` to `"authEnabled": false`. Then restart RackHD service.
+
+.. code::
+
+   vagrant@rackhd:~$ sudo service rackhd restart
+
+-  Wait several seconds, and then type the following RackHD restful API to ensure RackHD is started and is running successfully.
+
+.. code::
+
+    vagrant@rackhd:~$ curl localhost:8080/api/2.0/nodes
+
+-  The following message appears when RackHD is booting and not ready. The process takes several seconds. You can retry after 10 seconds.
+
+.. code::
+
+  curl: (7) Failed to connect to localhost port 8080: Connection refused
+
+-  When the API returns "[]"(no nodes discovered so far) RackHD software stacks are running successfully.
+
+Discovery
+----------
+Vnode is used to do node discovery. A vnode is defined in the file ``Vagrantfile``. The infrasim/quanta_d51 base box is used to set up a vnode. If you want to learn more about Infrasim, you can go to https://github.com/InfraSIM. UltraVNC Viewer can be used to view the progress of node discovery. In this environment, vnode work in the port ``15901`` instead of ``5901``.
```

Comment:
The vnode work in the port 15901. In vagrant box, port mapping is used. The port 5901  is mapped to 15901